### DISCUSSION TITLE
Adjoint equations failed to converge

### DISCUSSION THREAD

**Discussion by eduVital** [mdolab/dafoam#351] [General]
Hello,
I'm running a K-omega SST incompressible simulation on a NACA 0012, mach 0.2 and AoA 2. The primal converges well but the adjoint don't - the residue stays pratically constant. All checkMeshs pass, solver is DASimpleFoam and div(phi, U) scheme was alway to upwind.
I tried checking the website and previous threads, in a way that I:

- Increased **pcFillLevel** to 2 or using 'nd' as  **jacMatReOrdering** method.
- velocity-inlet + static pressure-outlet boundary conditions.
- Change adjStateOrdering to cell and change jacMatReOrdering to natural.
- Increase the y+ to 35.
- Use pcFillLevel = 2.

The log:
`ILU PC Fill Level: 2
GMRES Max Iterations: 1000
GMRES Relative Tolerance: 1e-06
GMRES Absolute Tolerance: 1e-14
Calculating [dRdW]^T * Psi using reverse-mode AD
Solving Linear Equation... 1501 s
Main iteration 0 KSP Residual norm 1.062531718164e-01 1684 s
Main iteration 100 KSP Residual norm 1.062531718164e-01 1792 s
Main iteration 200 KSP Residual norm 1.062531718164e-01 1905 s
Main iteration 300 KSP Residual norm 1.062531718164e-01 2025 s
Main iteration 400 KSP Residual norm 1.062531718164e-01 2151 s
Main iteration 500 KSP Residual norm 1.062531718164e-01 2284 s
Main iteration 600 KSP Residual norm 1.062531718164e-01 2425 s
Main iteration 700 KSP Residual norm 1.062531718164e-01 2575 s
Main iteration 800 KSP Residual norm 1.062531718164e-01 2733 s
Main iteration 900 KSP Residual norm 1.062531718164e-01 2897 s
Main iteration 1000 KSP Residual norm 1.062531718164e-01 3067 s
Main iteration 1000 KSP Residual norm 1.062531718164e-01 3068 s 
Total iterations 1000
Solving Linear Equation... Completed! 3068 s
Residual tolerance not satisfied, solution failed!
Calculating [dRdXv]^T * Psi using reverse-mode AD
Calculating [dRdAOA]^T * Psi using reverse-mode AD
OrderedDict([(('cruise.aero_post.CL', 'dvs.shape'), array([[-0.0003284889468748, -0.0003284889421905, -0.0059703227551462,
        -0.0059703227503068,  0.0010812581729592,  0.001081258181581 ,
         0.0028760538638195,  0.0028760538769217, -0.0005291961003292,
        -0.0005291961053856,  0.0019684683572493,  0.001968468433222 ,
        -0.000642492476355 , -0.0006424925536518,  0.0015640512466151,
         0.0015640516053072, -0.0004462903378834, -0.0004462906278417,
         0.0004269589731919,  0.0004269588850872]])), (('cruise.aero_post.CL', 'dvs.aoa'), array([[-0.0000000000000042]]))])
Finalising parallel run
Finalising parallel run`

files: 
[runScript.txt](https://github.com/mdolab/dafoam/files/9609348/runScript.txt)
[genFFD.txt](https://github.com/mdolab/dafoam/files/9609350/genFFD.txt)
[controlDict (2).txt](https://github.com/mdolab/dafoam/files/9609355/controlDict.2.txt)

**Comment by friedenhe** [comment]
Have you tried to use the SST model to run the tutorial case without anychange? If that works, I suggest you change the tutorial setup to your setup one step at a time, and you can figure out which change breaks the case.
